docs/CONFIG.md

```diff
@@ -9,7 +9,8 @@ Configuration lives in [config.yml](../config.yml).
   - `api_key`: The API key for Elasticsearch connection. You can't set `api_key` when basic auth is used.
   - `ssl`: Whether SSL is used for the Elasticsearch connection.
   - `ca_certs`: Path to a CA bundle.
-  - `bulk`: Options for the Bulk API calls behavior
+  - `bulk`: Options for the Bulk API calls behavior - all options can be
+    overriden by each source class
```

Comment:
It doesn't feel right to override config by code.

If we want to override by source type, shouldn't we have something like this in config?

```
bulk:
    ...
    overrides:
        s3:
            ...
        mysql:
            ...
```

Reply:
It offers the ability for a user to change the value in the connector configuration in Kibana, and set the default value when a connector is added.

In our source classes, `get_default_configuration` is called when we do the initial setup of a new connector to define the default configuration. If a source needs to set another default for one of these parameters, this will be the place where it's possible..

If it's done in the yaml file, we won't be able for native connectors to tweak that behavior. The use case is when a customer want to reduce or augment the number of concurrent connection to its backend